response/response.go

```diff
@@ -258,30 +258,28 @@ func (d *Data) UnmarshalJSON(b []byte) error {
 	}
 
 	data, ok := d.Content.(map[string]interface{})
-	if ok {
-		// count how many collections were provided
-		var count int
-		for _, value := range data {
-			if _, ok := value.(map[string]interface{}); ok {
-				count++
-			}
-			if _, ok := value.([]interface{}); ok {
-				count++
-			}
-		}
-		if count > 1 {
-			// we can stop there since this is not a single collection
-			return nil
+	if !ok {
+		return nil
+	}
+	// count how many collections were provided
+	var count int
+	for _, value := range data {
+		switch value.(type) {
+		case map[string]interface{}, []interface{}:
+			count++
 		}
-
-		for key, value := range data {
-			if _, ok := value.(map[string]interface{}); ok {
-				d.Type = key
-				d.Content = data[key]
-			} else if _, ok := value.([]interface{}); ok {
-				d.Type = key
-				d.Content = data[key]
-			}
+	}
+	if count > 1 {
+		// we can stop there since this is not a single collection
+		return nil
+	}
+	for key, value := range data {
+		if _, ok := value.(map[string]interface{}); ok {
```

Comment:
Maybe a bit cleaner (DRYer at least) to do a type switch like we have above:

```
switch value.(type) {
  case map[string]interface{}, []interface{}:
    d.Type = key
    d.Content = data[key]
}
```

Reply:
done